android-pdf-viewer/src/main/java/com/github/barteksc/pdfviewer/link/DefaultLinkHandler.kt

```diff
@@ -20,16 +20,44 @@ package com.github.barteksc.pdfviewer.link
  */
 import android.util.Log
 import com.github.barteksc.pdfviewer.PDFView
+import com.github.barteksc.pdfviewer.annotation.core.pdf.DocumentationOpener
+import com.github.barteksc.pdfviewer.annotation.core.pdf.PdfUtil
+import com.github.barteksc.pdfviewer.annotation.core.shapes.checkIfPointIsInsideRect
 import com.github.barteksc.pdfviewer.model.LinkTapEvent
 
-class DefaultLinkHandler(private val pdfView: PDFView) : LinkHandler {
-     override fun handleLinkEvent(event: LinkTapEvent) {
-        Log.i(TAG, "handleLinkEvent - event ¬--> X: " + event.originalX + " | Y: " + event.originalY)
-        Log.i(TAG, "--------------------------------------------------")
+class DefaultLinkHandler(
+    private val pdfView: PDFView,
+    private val pdfFilePath: String,
+    private val opener: DocumentationOpener
+) :
+    LinkHandler {
+    override fun handleLinkEvent(event: LinkTapEvent) {
+        Log.i(
+            TAG,
+            "handleLinkEvent - event ¬--> X: " + event.originalX + " | Y: " + event.originalY
+        )
 
         val pdfPoint = pdfView.convertScreenPintsToPdfCoordinates(event.originalX, event.originalY)
         Log.i(TAG, "handleLinkEvent - pdfPoint --> X: " + pdfPoint.x + " | Y: " + pdfPoint.y)
-        Log.i(TAG, "--------------------------------------------------")
+
+        val extractedAnnotations = PdfUtil.getAnnotationsFrom(pdfFilePath, pageNum = 1)
+        extractedAnnotations.forEach { annotation ->
+            val isAnnotationMatchFound = checkIfPointIsInsideRect(
+                pdfPoint,
+                topRight = annotation.points[1],
+                bottomLeft = annotation.points[3]
+            )
+            val documentation = annotation.relations?.documentation?.get(0)
+            if (isAnnotationMatchFound) {
+                if (documentation != null) {
+                    opener.openDocumentation(
+                        annotation.relations.documentation[0].schemaId,
+                        annotation.relations.documentation[0].documentId
+                    )
+                    return@forEach
+                }
+            }
+        }
```

Comment:
This is more complex than it needs to be. I would refactor it to

```kotlin
val clickedAnnotationannotations.firstOrNull { annotation->
    checkIfPointIsInsideRect(pdfPoint, annotation)
} ?: return
opener.openDocumentation(annotation.relations.documentation)
```

Reply:
Used `let` to continue with the annotation if not null

Reply:
> Used let to continue with the annotation if not null

Thats not really a good usecase for let. You are continuing even when annotation is null so I'm wondering if it makes sense to have a let block in the first place.

![2024-02-12_18-26](https://github.com/VERO-Digital-Solutions/AndroidPdfViewerV2/assets/14945479/96009296-6ec7-4d90-82e1-fb04ecf3fa52)

Reply:
Done like in the snippet